### ISSUE TITLE
Hang when selecting anything in menu

### ISSUE THREAD

**Issue by rbuzzell** [KillBait/AutoDrive_Course_Editor#5]
The program will hang when selecting anything in the menu.
I can only end it was a ctrl+c.
```
PS D:\Downloads> java -jar .\AutoDriveEditor.jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-12-21 14:07:47.126 [main] AutoDriveEditor - Java Runtime Version 16
[INFO ] 2021-12-21 14:07:47.438 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2021-12-21 14:07:47.439 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor.....
[INFO ] 2021-12-21 14:07:47.657 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2021-12-21 14:07:47.658 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2021-12-21 14:07:47.658 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2021-12-21 14:07:50.157 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Load Config
PS D:\Downloads> java -jar .\AutoDriveEditor.jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-12-21 14:10:58.064 [main] AutoDriveEditor - Java Runtime Version 16
[INFO ] 2021-12-21 14:10:58.357 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2021-12-21 14:10:58.358 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor.....
[INFO ] 2021-12-21 14:10:58.584 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2021-12-21 14:10:58.584 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2021-12-21 14:10:58.584 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2021-12-21 14:11:18.848 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: 4x
PS D:\Downloads> java --version
openjdk 16.0.2 2021-07-20
OpenJDK Runtime Environment Temurin-16.0.2+7 (build 16.0.2+7)
OpenJDK 64-Bit Server VM Temurin-16.0.2+7 (build 16.0.2+7, mixed mode, sharing)
```

**Comment by KillBait** [comment]
I have to be honest, i;ve never seen or heard of this, so this has me perplexed. Could you try these steps

1 - try running the editor with 'java.exe -jar -Xms1024m -Xmx1024m .\AutoDriveEditor.Jar'
2 - run the editor, don't go near the menu's and use the load config shortcut Alt-l

If that doesn't work, I've linked a different (currently unreleased) version, try the same steps with this version

https://github.com/KillBait/AutoDrive_Course_Editor/raw/refactor/AutoDriveEditor.jar

Thanks for the report

**Comment by rbuzzell** [comment]
Same issue trying both of those.

Is it possible this is happening because I'm using an OpenJDK from adoptium instead of an Oracle release?

Is there any other information I can provide?

**Comment by KillBait** [comment]
I have been doing some testing and searching but come up empty, see attached pic.

![JDK16](https://user-images.githubusercontent.com/7914589/147140539-a07fa7fe-fa9e-4f92-b950-f5fa7ecd2ec1.png)

I always compile and test against the minimum JDK version 13, i do know that opening the first file chooser or dialog box can have a second or two pause before it shows, but subsequent open times are near instant.

If you know java, feel free to look over the source code, as far as i am aware, i'm not doing anything out of the ordinary.

My only suggestion at the moment is to try a different java distribution, i'm assuming you are trying to run this on a windows
machine?

**Comment by rbuzzell** [comment]
Hmm, a few seconds for you but forever for me lol. Oracle's Java 17 doesn't seem to do it either, and the OpenJDK 15 I started with before going to OpenJDK 17 _also_ didn't seem to work.

I can give it a look but Java GUIs are _not_ my normal forte.

**Comment by KillBait** [comment]
There is one other suggestion i  can think of.

I started to update the editor around 13th August this year, all the commits i have done are listed [here](https://github.com/KillBait/FS19_AutoDrive/commits/beta-testing?after=c5af4e1375c8af7548d5477c70fb9443ac2d60b0+34&branch=beta-testing), almost every commit i did had a JAR included, maybe try a few and see if the same issue is present, at least that might give me a place to start from

**Comment by rbuzzell** [comment]
Oh! Yeah, this is on Windows 10. 

Hmm. Seems to come to a complete halt on my machine in EditorListener.java on the line

`JFileChooser fc = new JFileChooser();`

Googling jFileChooser suggests I may have hit an edge case with it somehow.

**Comment by KillBait** [comment]
just to confirm its that line of code, can you try this version?, i put extra logging to see where it freezes on a config load

[AutoDriveEditor.zip](https://github.com/KillBait/AutoDrive_Course_Editor/files/7765515/AutoDriveEditor.zip)

you should see at least one of these

`LOG.info("creating load config filechooser");`
` JFileChooser loadConfigFileChooser = new JFileChooser();`
`LOG.info("created load config filechooser");`

The attached editor is a severely cutdown version for testing

**Comment by rbuzzell** [comment]
Yep, freezes right at this point.

```
PS D:\Downloads> java -jar -Xms1024m -Xmx1024m .\AutoDriveEditor.Jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-12-22 20:43:53.002 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor v0.51.0 .....
[INFO ] 2021-12-22 20:43:53.003 [AWT-EventQueue-0] AutoDriveEditor - Java Runtime Version 17
[INFO ] 2021-12-22 20:43:53.008 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2021-12-22 20:43:53.234 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2021-12-22 20:43:53.234 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2021-12-22 20:43:53.234 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2021-12-22 20:43:53.358 [AWT-EventQueue-0] AutoDriveEditor - Connecting to GitHub for update check
[INFO ] 2021-12-22 20:43:53.990 [AWT-EventQueue-0] AutoDriveEditor - No update available... Remote version 0.51.0 matches current version
[INFO ] 2021-12-22 20:43:53.991 [AWT-EventQueue-0] AutoDriveEditor - Setup Complete....
[INFO ] 2021-12-22 20:43:56.850 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Load Config
[INFO ] 2021-12-22 20:43:56.850 [AWT-EventQueue-0] AutoDriveEditor - creating load config filechooser
```

**Comment by KillBait** [comment]
That confirms it. I spent a few hours searching and the only thing i came up with is filechoosers have issues with
network shares. I'm well aware that java Swing components are not thread safe and need to run on the event
dispatch thread, from the log it looks like that is happening.

i could try a wrap the filechooser code in a `EventQueue.invokeLater(new Runnable() {})` just to make sure it runs on
the event dispatch thread, but i think that is a long shot at best

From what i have been reading there is no good alternate to FileChoosers except (maybe) [FileChooser2](https://github.com/fracpete/vfsjfilechooser2) , but i have no idea
how this would behave on non windows machines

**Comment by KillBait** [comment]
ok.. i've attached another version.

i've wrapped the config load in a Runnable() to make 100% sure it on the the correct thread and
i've directly specified a directory for the filechooser on initialization, just to make sure its not the FileSystemView
that's messing things up

[AutoDriveEditor.zip](https://github.com/KillBait/AutoDrive_Course_Editor/files/7768739/AutoDriveEditor.zip)

**Comment by rbuzzell** [comment]
Ah! [I left it running last time](https://github.com/KillBait/AutoDrive_Course_Editor/issues/5#issuecomment-999985323) while I went to run some Christmas errands. Looks like it takes 61 minutes from "creating load filechooser" to "created load filechooser".

I'll give the version you [just posted](https://github.com/KillBait/AutoDrive_Course_Editor/issues/5#issuecomment-1000208914) a go and see if it's any better.

```
PS D:\Downloads> java -jar -Xms1024m -Xmx1024m .\AutoDriveEditor.Jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-12-22 20:43:53.002 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor v0.51.0 .....
[INFO ] 2021-12-22 20:43:53.003 [AWT-EventQueue-0] AutoDriveEditor - Java Runtime Version 17
[INFO ] 2021-12-22 20:43:53.008 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2021-12-22 20:43:53.234 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2021-12-22 20:43:53.234 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2021-12-22 20:43:53.234 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2021-12-22 20:43:53.358 [AWT-EventQueue-0] AutoDriveEditor - Connecting to GitHub for update check
[INFO ] 2021-12-22 20:43:53.990 [AWT-EventQueue-0] AutoDriveEditor - No update available... Remote version 0.51.0 matches current version
[INFO ] 2021-12-22 20:43:53.991 [AWT-EventQueue-0] AutoDriveEditor - Setup Complete....
[INFO ] 2021-12-22 20:43:56.850 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Load Config
[INFO ] 2021-12-22 20:43:56.850 [AWT-EventQueue-0] AutoDriveEditor - creating load config filechooser
[INFO ] 2021-12-22 21:44:42.163 [AWT-EventQueue-0] AutoDriveEditor - created load config filechooser
[INFO ] 2021-12-22 21:44:42.163 [AWT-EventQueue-0] AutoDriveEditor - checked isStale
[INFO ] 2021-12-22 21:44:42.166 [AWT-EventQueue-0] AutoDriveEditor - opening filechooser
[INFO ] 2021-12-24 13:22:57.783 [AWT-EventQueue-0] AutoDriveEditor - Stopping ConnectionDraw Thread
[INFO ] 2021-12-24 13:22:57.783 [AWT-EventQueue-0] AutoDriveEditor - Stopping NodeDraw thread
[INFO ] 2021-12-24 13:22:57.783 [ConnectionDraw Thread] AutoDriveEditor - ConnectionDraw Thread exiting
[INFO ] 2021-12-24 13:22:57.784 [NodeDraw Thread] AutoDriveEditor - NodeDraw Thread exiting
[INFO ] 2021-12-24 13:22:57.835 [AWT-EventQueue-0] AutoDriveEditor - Completed saving Editor Config
```

**Comment by rbuzzell** [comment]
The version you posted most recently appears to get a _tiny_ bit further then hold at the same spot.

```
PS D:\Downloads> java -jar -Xms1024m -Xmx1024m .\AutoDriveEditor.Jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-12-24 13:27:28.444 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor v0.51.0 .....
[INFO ] 2021-12-24 13:27:28.446 [AWT-EventQueue-0] AutoDriveEditor - Java Runtime Version 17
[INFO ] 2021-12-24 13:27:28.450 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2021-12-24 13:27:28.677 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2021-12-24 13:27:28.677 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2021-12-24 13:27:28.678 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2021-12-24 13:27:28.801 [AWT-EventQueue-0] AutoDriveEditor - Connecting to GitHub for update check
[INFO ] 2021-12-24 13:27:29.540 [AWT-EventQueue-0] AutoDriveEditor - No update available... Remote version 0.51.0 matches current version
[INFO ] 2021-12-24 13:27:29.540 [AWT-EventQueue-0] AutoDriveEditor - Setup Complete....
[INFO ] 2021-12-24 13:27:31.307 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Load Config
[INFO ] 2021-12-24 13:27:31.313 [AWT-EventQueue-0] AutoDriveEditor - creating load config filechooser
[INFO ] 2021-12-24 13:27:31.316 [AWT-EventQueue-0] AutoDriveEditor - JAR Path : /D:/Downloads/AutoDriveEditor.jar
[INFO ] 2021-12-24 13:27:31.317 [AWT-EventQueue-0] AutoDriveEditor - Path : /D:/Downloads/
```

**Comment by KillBait** [comment]
It still freezing at the same place, the extra logging was from detecting the launch folder before creating the filechooser. I'm
really stumped as to why it's freezing, i've made sure it's on the correct thread, changed the initial directory, changed
around a load of things and still no go, it must be a very strange edge case.

I've attached another test JAR ( sorry about this ), this one uses a filechooser alternative (VFSJFileChooser), i've only replaced
the load config dialog, all others still use the regular filechooser. Hopefully it will work for you.

**Comment by rbuzzell** [comment]
That seems to work, though I changed my end as well.

I explored [this](https://github.com/KillBait/AutoDrive_Course_Editor/issues/5#issuecomment-1000194818) as having an issue with the _existence_ of a fileshare even if it wasn't loading into one and removed a vestigial pair I had, and that seems to have unhung this.

I'm willing to call this an issue on my end or somewhere upsteam in Java itself. I was off in some weird edge case that I have doubts any other person will encounter. If anyone does: Using Windows 10 Pro I had both an NFS and an SMB share mapped, though unfortunately I didn't test to see which one unhung it when removed.

**Comment by KillBait** [comment]
Its good to hear you found the cause of the issue, i apologize for putting you through a bunch of testing, i was not aware
of this network share issue before you posted and started searching around the internet.

At least i can add this to the list of known issues.

**Comment by yann-mrt** [comment]
Good day,
Not sure if it is the same issue or different, but it is hang as well when for example I click on the icon to "create a curve" then click on one node (like to start the curve), but want to click on the icon to "delete a node" because I realize I select the wrong action/icon (without to deselect the node) course editor hand and can't recover. 
This is the second time it happen to me but with different action/icon but always the same way : forgot to deselect the node before to click on the new action/icon.
Using JDK 17 - course editor 0.60.2 - Win 11

C:\Users\Yannick\Documents\My Games\FarmingSimulator2022\AutoDrive Course Editor>java.exe -jar -Xms1024m -Xmx1024m .\AutoDriveEditor.Jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2022-01-06 23:00:16.812 [AWT-EventQueue-0] AutoDriveEditor - Starting AutoDrive Editor v0.60.2 .....
[INFO ] 2022-01-06 23:00:16.813 [AWT-EventQueue-0] AutoDriveEditor - Java Runtime Version 17
[INFO ] 2022-01-06 23:00:16.817 [AWT-EventQueue-0] AutoDriveEditor - 'AutoDriveEditor_en_US.properties' loaded
[INFO ] 2022-01-06 23:00:16.979 [AWT-EventQueue-0] AutoDriveEditor - Initializing CopyPaste Manager
[INFO ] 2022-01-06 23:00:16.979 [ConnectionDraw Thread] AutoDriveEditor - Starting ConnectionDraw Thread
[INFO ] 2022-01-06 23:00:16.979 [NodeDraw Thread] AutoDriveEditor - Starting NodeDraw thread
[INFO ] 2022-01-06 23:00:17.145 [AWT-EventQueue-0] AutoDriveEditor - Connecting to GitHub for update check[INFO ] 2022-01-06 23:00:17.914 [AWT-EventQueue-0] AutoDriveEditor - No update available... Remote version 0.60.2 matches current version
[INFO ] 2022-01-06 23:00:22.246 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Load Config
[INFO ] 2022-01-06 23:00:30.857 [AWT-EventQueue-0] AutoDriveEditor - loadFile: C:\Users\Yannick\Documents\My Games\FarmingSimulator2022\AutoDrive Course Editor\Elmatador Yann\AutoDrive_config.xml
[INFO ] 2022-01-06 23:00:30.876 [AWT-EventQueue-0] AutoDriveEditor - Root element :AutoDrive
[INFO ] 2022-01-06 23:00:30.876 [AWT-EventQueue-0] AutoDriveEditor - FS22 Config detected
[INFO ] 2022-01-06 23:00:30.877 [AWT-EventQueue-0] AutoDriveEditor - Config was saved by AutoDrive '2.0.0.2'
[INFO ] 2022-01-06 23:00:30.879 [AWT-EventQueue-0] AutoDriveEditor - Current Element :waypoints
[INFO ] 2022-01-06 23:00:31.637 [AWT-EventQueue-0] AutoDriveEditor - Loaded config for map : Elm_Creek_Edit_By_Stevie_
[INFO ] 2022-01-06 23:00:31.637 [AWT-EventQueue-0] AutoDriveEditor - Config load complete.
[INFO ] 2022-01-06 23:00:31.639 [AWT-EventQueue-0] AutoDriveEditor - failed to load map image from JAR .. trying alternate locations
[INFO ] 2022-01-06 23:00:31.693 [AWT-EventQueue-0] AutoDriveEditor - Loaded mapImage from /C:/Users/Yannick/Documents/My Games/FarmingSimulator2022/AutoDrive Course Editor/mapImages/Elm_Creek_Edit_By_Stevie_.png
[INFO ] 2022-01-06 23:00:31.693 [AWT-EventQueue-0] AutoDriveEditor - Selected Image size is 2048 x 2048
[INFO ] 2022-01-06 23:00:31.705 [AWT-EventQueue-0] AutoDriveEditor - HeightMap path = C:\Users\Yannick\Documents\My Games\FarmingSimulator2022\AutoDrive Course Editor\Elmatador Yann\terrain.heightmap.png
[INFO ] 2022-01-06 23:00:33.254 [AWT-EventQueue-0] AutoDriveEditor - Failed to load HeightMap
javax.imageio.IIOException: Can't read input file!
        at java.desktop/javax.imageio.ImageIO.read(ImageIO.java:1310)
        at AutoDriveEditor.MapPanel.MapImage.loadHeightMap(MapImage.java:165)
        at AutoDriveEditor.Listeners.MenuListener.actionPerformed(MenuListener.java:56)
        at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
        at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
        at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
        at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
        at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374)
        at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028)
        at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072)
        at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
        at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
        at java.desktop/java.awt.Component.processEvent(Component.java:6391)
        at java.desktop/java.awt.Container.processEvent(Container.java:2266)
        at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
        at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
        at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
        at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
        at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
        at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
        at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
        at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
        at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
        at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
        at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
[INFO ] 2022-01-06 23:01:42.245 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Remove Nodes
[INFO ] 2022-01-06 23:02:00.147 [AWT-EventQueue-0] AutoDriveEditor - Rectangle started at 899/433
[INFO ] 2022-01-06 23:02:01.730 [AWT-EventQueue-0] AutoDriveEditor - Rectangle ended at 1053/108
[INFO ] 2022-01-06 23:02:01.732 [AWT-EventQueue-0] AutoDriveEditor - Removing all nodes in area
[INFO ] 2022-01-06 23:02:02.417 [AWT-EventQueue-0] AutoDriveEditor - Rectangle started at 1047/160
[INFO ] 2022-01-06 23:02:04.350 [AWT-EventQueue-0] AutoDriveEditor - Rectangle ended at 1565/343
[INFO ] 2022-01-06 23:02:04.353 [AWT-EventQueue-0] AutoDriveEditor - Removing all nodes in area
[INFO ] 2022-01-06 23:02:16.775 [AWT-EventQueue-0] AutoDriveEditor - Rectangle started at 941/659
...
...
[INFO ] 2022-01-06 23:52:07.738 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Remove Destinations
[INFO ] 2022-01-06 23:52:22.299 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Remove Nodes
[INFO ] 2022-01-06 23:52:25.219 [AWT-EventQueue-0] AutoDriveEditor - Rectangle started at 1138/355
[INFO ] 2022-01-06 23:52:28.766 [AWT-EventQueue-0] AutoDriveEditor - Rectangle ended at 489/564
[INFO ] 2022-01-06 23:52:28.766 [AWT-EventQueue-0] AutoDriveEditor - Removing all nodes in area
[INFO ] 2022-01-06 23:52:36.598 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Create Secondary Node
[INFO ] 2022-01-06 23:52:49.944 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Connect Nodes
[INFO ] 2022-01-06 23:52:59.067 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Quadratic Bezier
[INFO ] 2022-01-06 23:53:05.207 [AWT-EventQueue-0] AutoDriveEditor - ActionCommand: Confirm Curve
[INFO ] 2022-01-06 23:53:13.816 [NodeDraw Thread] GlobalExceptionHandler - Cannot invoke "java.awt.Point.getX()" because the return value of "AutoDriveEditor.MapPanel.MapPanel.getMousePosition()" is null

java.lang.NullPointerException: Cannot invoke "java.awt.Point.getX()" because the return value of "AutoDriveEditor.MapPanel.MapPanel.getMousePosition()" is null
        at AutoDriveEditor.MapPanel.MapPanel$NodeDrawThread.run(MapPanel.java:339) ~[AutoDriveEditor.jar:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]

![image](https://user-images.githubusercontent.com/74056604/148470065-725914d4-10f8-45ea-8e94-e75d0c5242e2.png)
(The arrow on the screen shot, if from the course editor, i didn't add anything)
Sorry if this is the same as post above.

Thanks
Yann

**Comment by KillBait** [comment]
Sorry @yann-mrt, i missed this post, this crashed is already fixed in v0.70.0
